aws_mp_utils/scripts/cli_utils.py

```diff
@@ -157,3 +157,65 @@ def get_mp_client(profile, region):
         region,
         session
     )
+
+
+# -----------------------------------------------------------------------------
+def ingress_rule_repl():
+    """
+    Read eval and print loop to get a list of ingress rules
+
+    The rules are added to automatically created security groups.
+    """
+    rules = []
+    while True:
+        rule = {
+            'ip_ranges': []
+        }
+        if click.confirm('Add an ingress rule?'):
+            rule['FromPort'] = click.prompt(
+                'Enter the source port ',
+                type=click.IntRange(min=1, max=65535),
+                default=22
+            )
+            rule['ToPort'] = click.prompt(
+                'Enter the destination port ',
+                type=click.IntRange(min=1, max=65535),
+                default=22
+            )
+            rule['IpProtocol'] = click.prompt(
+                'Enter the IP protocol (tcp or udp)',
+                type=click.Choice(['tcp', 'udp']),
+                default='tcp'
+            )
+            rule['ip_ranges'] = ip_range_repl()
+            rules.append(rule)
+        else:
+            break
+
+    return rules
+
+
+def ip_range_repl():
+    """
+    Read eval and print loop to get a list of ip ranges
+
+    The ip ranges are added to the ingress rule.
+    """
+    ip_ranges = [ip_range_prompt()]
+
+    while True:
+        if click.confirm('Add another IP range?'):
+            ip_ranges.append(ip_range_prompt())
+        else:
+            break
+
+    return ip_ranges
+
+
+def ip_range_prompt():
+    """Prompt for an IP range and return the user input."""
+    return click.prompt(
+        'Enter an IP range (CIDR format xxx.xxx.xxx.xxx/nn)',
+        type=str,
+        default='0.0.0.0/0'
```

Comment:
Should also support IPv6, but that can be added in another PR.

Reply:
This still handles IPv6. It just defaults to IPv4. It is treated as a string. We can add more validation if we want to confirm the input is valid.